recipes/meson/all/conanfile.py

```diff
@@ -20,6 +20,12 @@ class MesonInstallerConan(ConanFile):
 exec "$meson_dir/meson.py" "$@"
 """
 
+    def requirements(self):
+        self.requires("ninja/1.10.1")
+
+    def package_id(self):
```

Comment:
Any particular reason for moving this up? We pretty much always have this as the last method

Reply:
To follow execution order. package_id is called after requirements, so it's easier to understand recipe's logic (here it's simple, I agree).

Reply:
I think having this uniform across as many recipes as possible is more important than any particular order specifically. So unless you want to open an issue talking about a new standard and mass changing every recipe I would recommend against changing it for selected/arbitrary recipes 🤷‍♂️

Reply:
AFAIK, there are no guidelines in CCI regarding position of methods in conanfile, but most recipes follow order of executions for common methods like source, build, package, package_info.
A lot of recipes don't have package_id at the end, probably more than 50% having package_id. So there is no such thing as uniform across recipes.

Reply:
@SpaceIm 's motivation makes sense.
It's nice to have the same the methods listed in the same order as they are executed.

I would not make it a hard requirement though, just nice.

If we really want a uniform recipe, we should write a reformatter to reformat the recipes.
It should in no way become a rule/hook because that would scare away too many people.

Reply:
I'd say for me order is not important. you may keep alphanumeric order of methods, or use another one, if you want.
it just make a bit harder to review PRs if they are re-ordering methods constantly.